sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/ResourceManagerTestProxyTestBase.java

```diff
@@ -216,6 +228,106 @@ protected String clientIdFromFile() {
+                }
+                builder.redirectErrorStream(true);
+                Process process = builder.start();
+
+                StringBuilder output = new StringBuilder();
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream(),
+                    StandardCharsets.UTF_8))) {
+                    String line;
+                    while (true) {
+                        line = reader.readLine();
+                        if (line == null) {
+                            break;
+                        }
+
+                        if (windowsProcessErrorMessage.matcher(line).find()
+                            || shProcessErrorMessage.matcher(line).find()) {
+                            throw LOGGER.logExceptionAsError(new RuntimeException("AzureCliCredential authentication unavailable. Azure CLI not installed."
+                                + "To mitigate this issue, please refer to the troubleshooting guidelines here at "
+                                + "https://aka.ms/azsdk/java/identity/azclicredential/troubleshoot"));
+                        }
+                        output.append(line);
+                    }
+                }
+                String processOutput = output.toString();
+
+                // wait(at most) 10 seconds for the process to complete
+                process.waitFor(10, TimeUnit.SECONDS);
+
+                if (process.exitValue() != 0) {
+                    if (processOutput.length() > 0) {
+                        if (processOutput.contains("az login") || processOutput.contains("az account set")) {
+                            throw LOGGER.logExceptionAsError(new RuntimeException("AzureCliCredential authentication unavailable. Azure CLI not installed."
+                                + "To mitigate this issue, please refer to the troubleshooting guidelines here at "
+                                + "https://aka.ms/azsdk/java/identity/azclicredential/troubleshoot"));
+                        }
+                        throw LOGGER.logExceptionAsError(new ClientAuthenticationException("get Azure CLI current signed-in user failed", null));
+                    } else {
+                        throw LOGGER.logExceptionAsError(
+                            new ClientAuthenticationException("Failed to invoke Azure CLI ", null));
+                    }
+                }
+
+                LOGGER.verbose("Get Azure CLI signed-in user => A response was received from Azure CLI, deserializing the"
+                    + " response into an signed-in user.");
+                try (JsonReader reader = JsonProviders.createReader(processOutput)) {
+                    Map<String, Object> signedInUserInfo = reader.readMap(JsonReader::readUntyped);
+                    String userPrincipalName = (String) signedInUserInfo.get("userPrincipalName");
```

Comment:
see if you need `@Suppress` on method, on cast.

Reply:
CI/IDE didn't report issue. Guess it's fine.